lib/dest/write-contents/write-stream.js

```diff
@@ -41,8 +60,10 @@ function writeStream(file, optResolver, onWritten) {
     file.contents.removeListener('error', onComplete);
 
     // TODO: this is doing sync stuff & the callback seems unnecessary
-    // TODO: Replace the contents stream or use a clone?
-    readStream(file, complete);
+    readStream(file, createResolver(readConfig, {
```

Comment:
Yeah, ouch, indeed. I looked briefly into avoiding this whole mess -- i.e. for a shortcut to re-source a stream, rather than actually going through `src/read-contents/read-stream`...

But then I figured that even if I found something it probably shouldn't be in this PR. The current master branch does it this way, albeit without the added complexity of the option resolver. So I propose we look into this issue separately?